pkg/controllers/vrep/replication_reconciler.go

```diff
@@ -238,33 +237,92 @@ func (r *ReplicationReconciler) checkSandboxExists() error {
 	return nil
 }
 
-// choose the source host and target host
-// (first host where db is up in the specified cluster)
-func (r *ReplicationReconciler) determineSourceAndTargetHosts() (err error) {
-	// assume source could be read-only, no subcluster constraints
-	upPodIP, ok := r.SourcePFacts.FindFirstUpPodIP(true, "")
-	if !ok {
-		err = fmt.Errorf("cannot find any up hosts in source database cluster")
+func (r *ReplicationReconciler) validateServiceIP(ctx context.Context, svcName types.NamespacedName) error {
+	srv := corev1.Service{}
+	if err := r.VRec.Client.Get(ctx, svcName, &srv); err != nil {
+		return err
+	}
+	if srv.Spec.Type == "ExternalName" {
+		return fmt.Errorf("cannot use serviceName to specify an ExternalName service")
+	}
+	if srv.Spec.ClusterIP == "" {
+		return fmt.Errorf("%s does not have an assigned ClusterIP", svcName.String())
+	}
+	return nil
+}
+
+// used below to avoid a really messy if condition
+func (r *ReplicationReconciler) shouldPickService(ctx context.Context, vdb *vapi.VerticaDB, sc *vapi.Subcluster,
+	sandboxName, svcName string) (should bool, err error) {
+	scSvcName := types.NamespacedName{Namespace: vdb.Namespace, Name: sc.ServiceName}
+	if scSvcName.Name == "" {
+		scSvcName = names.GenExtSvcName(vdb, sc)
+	}
+	if svcName != "" {
+		return scSvcName.Name == svcName, r.validateServiceIP(ctx, scSvcName)
+	}
+
+	if vdb.GetSubclusterSandboxName(sc.Name) != sandboxName {
```

Comment:
the reason i did it this way is so that specifying a service is enough to direct traffic to the sandbox. if you feel strongly that we shouldn't allow that behavior i'll reverse this